text/batch-l2.md

```diff
@@ -0,0 +1,1540 @@
+This will choose the lowest-cost instance type that fits all the jobs in the queue. If instances of that type are not available,
+the queue will not choose a new type; instead, it will wait for the instance to become available.
+This can stall your `Queue`, with your compute environment only using part of its max capacity (or none at all) until the `BEST_FIT` instance becomes available. 
+
+If you are running a workflow that does not tolerate interruptions and you want to maximize throughput, you can use `AllocationStrategy.BEST_FIT_PROGRESSIVE`.
+This strategy will examine the Jobs in the queue and choose whichever instance type meets the requirements of the jobs in the queue and with the lowest cost per vCPU, just as `BEST_FIT`.
+However, if not all of the capacity can be filled with this instance type, it will choose a new next-best instance type to run any jobs that couldn’t fit into the `BEST_FIT` capacity.
+To make the most use of this allocation strategy, it is recommended to use as many instance classes as is feasible for your workload.
+This example shows a `ComputeEnvironment` that uses `BEST_FIT_PROGRESSIVE` with `InstanceType.OPTIMAL` and `InstanceClass.A1` instance types:
+
+```ts
+const computeEnv = new batch.ManagedEC2ComputeEnvironment(this, 'myEc2ComputeEnv', {
+   allocationStrategy: AllocationStrategy.BEST_FIT_PROGRESSIVE,
+  instanceTypes: [batch.InstanceType.OPTIMAL],
+  instanceClasses: [ec2.InstanceClass.A1],
+});
+```
+
+### Controlling vCPU allocation
+
+With a managed `ComputeEnvironment`, you can specify the maximum and minimum vCPUs it can have at any given time.
+You cannot do this with an unmanaged `ComputeEnvironment`, because you must provision and manage the instances yourself;
+that is, Batch will not scale them up and down as needed. This example shows how to configure these properties:
+
+```ts
+new batch.ManagedEC2ComputeEnvironment(this, 'myEc2ComputeEnv', {
+  instanceClasses: [ec2.InstanceClass.A1],
+  minvCPUs: 10,
+  maxvCpus: 100,
+});
+```
+
+This means that the `ComputeEnvironment` will always maintain 10 vCPUs worth of instances,
+even if there are no Jobs in the queues that link to this `ComputeEnvironment`.
+
+### Sharing a ComputeEnvironment between multiple JobQueues
+
+Multiple `JobQueue`s can share the same `ComputeEnvironment`.
+If multiple Queues are attempting to submit Jobs to the same `ComputeEnvironment`,
+Batch will pick the Job from the Queue with the highest priority.
+This example creates two `JobQueue`s that share a `ComputeEnvironment`:
+
+```ts
+const sharedComputeEnv = new batch.FargateComputeEnvironment(this, 'spotEnv', {
+  spot: true,
+});
+
+const lowPriorityQueue = new batch.JobQueue(this, 'JobQueue', {
+   priority: 1,
+});
+
+const highPriorityQueue = new batch.JobQueue(this, 'JobQueue', {
+   priority: 10,
+});
+
+lowPriorityQueue.addComputeEnvironment(sharedComputeEnv, 1);
+highPriorityQueue.addComputeEnvironment(sharedComputeEnv, 1);
+```
+
+### Fairshare Scheduling
```

Comment:
This whole thing is still clear as muck to me, but it's not clear to me that it can be cleared up anyway.

Reply:
I've completely reworked this section based off new information